tests/erlang_tests/test_binary_to_integer.erl

```diff
@@ -27,7 +27,8 @@ start() ->
         safe_list_to_integer({}) +
         safe_list_to_integer(<<"+10">>) - 10 + safe_list_to_integer(<<"-">>) - 5 +
         safe_list_to_integer(<<"+">>) - 5 +
-        safe_list_to_integer(<<"">>) - 5.
+        safe_list_to_integer(<<"">>) - 5 +
+        binary_to_integer(<<"Atom">>, 30) + binary_to_integer(<<"VM">>, 32).
```

Comment:
The + arithmetic syntax is deprecated I believe now that we have stacktraces, tests should rather return 0 now.

Also, we are missing some important tests here, such as :
```
   binary_to_integer(<<"10">>, -1),
   binary_to_integer(<<"10">>, 0),
   binary_to_integer(<<"10">>, 1),
   binary_to_integer(<<"10">>, 2),
   binary_to_integer(<<"10">>, 36),
   binary_to_integer(<<"10">>, 37),
   binary_to_integer(<<"">>, 10),
   binary_to_integer(<<"0A">>, 16),
   binary_to_integer(<<"0a">>, 16).
```

I believe we don't have the same behavior with BEAM for some of them. The tests are run against BEAM so it will be detected and enforced.

Reply:
I'm not sure how to rewrite the current test in a good manner. I'll add a separate file only for `binary_to_integer/2` instead.